module/metrics/transaction.go

```diff
@@ -12,19 +12,20 @@ import (
 )
 
 type TransactionCollector struct {
-	transactionTimings         mempool.TransactionTimings
-	log                        zerolog.Logger
-	logTimeToFinalized         bool
-	logTimeToExecuted          bool
-	logTimeToFinalizedExecuted bool
-	timeToFinalized            prometheus.Summary
-	timeToExecuted             prometheus.Summary
-	timeToFinalizedExecuted    prometheus.Summary
-	transactionSubmission      *prometheus.CounterVec
-	scriptExecutedDuration     *prometheus.HistogramVec
-	transactionResultDuration  *prometheus.HistogramVec
-	scriptSize                 prometheus.Histogram
-	transactionSize            prometheus.Histogram
+	transactionTimings             mempool.TransactionTimings
+	log                            zerolog.Logger
+	logTimeToFinalized             bool
+	logTimeToExecuted              bool
+	logTimeToFinalizedExecuted     bool
+	timeToFinalized                prometheus.Summary
+	timeToExecuted                 prometheus.Summary
+	timeToFinalizedExecuted        prometheus.Summary
+	transactionSubmission          *prometheus.CounterVec
+	transactionSize                prometheus.Histogram
+	scriptExecutedDuration         *prometheus.HistogramVec
+	scriptExecutionErrorOnExecutor *prometheus.CounterVec
+	scriptSize                     prometheus.Histogram
+	transactionResultDuration      *prometheus.HistogramVec
 }
 
```

Comment:
please add a interface check here so it's more obvious when there's a mismatch.
```golang
var _ metrics.BackendScriptsMetrics = (*TransactionCollector)(nil)
var _ metrics.TransactionMetrics = (*TransactionCollector)(nil)
```

It's a good idea to also include them for the noop implementation